src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitDataImporter.cs

```diff
@@ -45,9 +48,20 @@ public async Task Import(string userId, string googleFitId, CancellationToken ca
         {
             AuthTokensResponse tokensResponse;
 
-            tokensResponse = await _googleFitTokensService.RefreshToken(googleFitId, cancellationToken);
-            if (tokensResponse == default)
+            // Get user's info for LastSync date
+            _logger.LogInformation("Query userInfo for user: {0}, platformId: {1}", userId, googleFitId);
+            var user = await _usersTableRepository.GetById(userId, cancellationToken);
+
+            user.UpdateImportState(GoogleFitConstants.GoogleFitPlatformName, DataImportState.Importing);
+
+            try
             {
+                tokensResponse = await _googleFitTokensService.RefreshToken(googleFitId, cancellationToken);
+            }
+            catch (TokenRefreshException ex)
+            {
+                _logger.LogError(ex, ex.Message);
+                user.UpdateImportState(GoogleFitConstants.GoogleFitPlatformName, DataImportState.Unauthorized);
```

Comment:
Same here. You need to call `await _usersTableRepository.Update(user, cancellationToken);` or this state will never be saved to the repository. It might be worth creating a private method to handle saving the user so you can handle the case where the eTag changed.